config/samples/resources/edgecontainernodepool/edgecontainernodepool_v1beta1_nodepool.yaml

```diff
@@ -0,0 +1,54 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+apiVersion: edgecontainer.cnrm.cloud.google.com/v1beta1
+kind: EdgecontainerCluster
+metadata:
+  name: edgecontainercluster-local-control-plane
```

Comment:
You need format `name-dep` or the test will fail
Ref: https://cnrm.git.corp.google.com/cnrm/+/refs/heads/archived/master/README.Samples.md#dependency-naming-convention
Also, if this is the dep resource, should we swap the order of resource and its dep resource? i.e.
```
apiVersion: edgecontainer.cnrm.cloud.google.com/v1beta1
kind: EdgeContainerNodePool
<other fields...>
---
apiVersion: edgecontainer.cnrm.cloud.google.com/v1beta1
kind: EdgeContainerCluster
<other fields...>
```

Reply:
Thanks for pointing out the convention naming for resource dependency.

Regarding the resource order, should the cluster be created first then the node-pool, so cluster need to be listed before node-pool?

Reply:
The dependency resource should be created before the target resource. While, KCC's [reconciliation](https://cloud.google.com/config-connector/docs/concepts/reconciliation) will determine the order and create the dependencies first. I do not think the order in the configuration file would affect how KCC works. 
Also, I took a look at other samples yaml files, maybe it's better to separate the target resource and the dependency resource like this? https://github.com/GoogleCloudPlatform/k8s-config-connector/tree/master/config/samples/resources/bigquerytable

Reply:
Gotcha, thanks for pointing out